tests/approx_matching_ut.cpp

```diff
@@ -200,4 +200,52 @@ SIMPLE_UNIT_TEST_SUITE(ApproxMatchingTest) {
 			ACCEPTS("");
 		}
 	}
+
+	SIMPLE_UNIT_TEST(StressTest) {
+		ystring regexp = "^";
+		for (size_t letter = 0; letter < 10; ++letter) {
+			regexp += ystring(3, letter + 'a');
+		}
+		regexp += "$";
+
+		auto fsm = BuildFsm(regexp.data());
+
+		ystring changedRegexp = regexp.substr(1, regexp.size() - 2);
```

Comment:
IMHO
It could be better to name this variable `text`.
Moreover, `text` could be created first and `regexp` could be expressed using it. Eg.:
```
const ystring regexp = "^" + text + "$";
```